Assets/Shopify/UIToolkit/Themes/Interfaces/IThemeBase.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Shopify.UIToolkit {
+    using Shopify.Unity.SDK;
+
+    public interface IThemeBase {
+        void OnLoadingStarted();
+        void OnLoadingFinished();
+        void OnError(ShopifyError error);
+        void OnPurchaseStarted();
+        void OnPurchaseCancelled();
+        void OnPurchaseCompleted();
```

Comment:
These interfaces make it a lot easier to test the behaviours because we can mock them